File: agriTech/src/main/java/com/example/agriTech/api/PlantUserAPI.java
package com.example.agriTech.api;

import java.util.List;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.example.agriTech.dto.plantuser.PlantUserDTO;
import com.example.agriTech.service.PlantUserService;

@RestController
@RequestMapping("/api/myplant")
public class PlantUserAPI {
    private final PlantUserService plantUserService;

    public PlantUserAPI(PlantUserService plantUserService) {
        this.plantUserService = plantUserService;
    }
    @GetMapping("/detail/{id}")
    public ResponseEntity<PlantUserDTO> getDetail(@PathVariable Long id) {
        return ResponseEntity.ok(plantUserService.getMyPlantDetail(id));
    }
    @GetMapping("/{userId}")
public ResponseEntity<List<PlantUserDTO>> getMyPlants(@PathVariable Long userId) {
    return ResponseEntity.ok(plantUserService.getMyPlantList(userId));
}

}

File: agriTech(DA3K)/app/src/main/java/com/example/agritechda3k/api/service/PlantUserApi.kt
package com.example.agritechda3k.api.service

import com.example.agritechda3k.api.dto.PlantUserDTO
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path

interface PlantUserApi {
    @GET("myplant/{userId}")
    suspend fun getMyPlantList(
        @Path("userId") userId: Long
    ): Response<List<PlantUserDTO>> // Phải tạo thêm file PlantUserDTO này
    @GET("myplant/detail/{id}")
    suspend fun getDetailMyPlant(@Path("id") id: Long) : Response<PlantUserDTO>

    // (Tiện tay viết luôn) API lấy danh sách device rảnh nếu ông cần
    @GET("api/devices/available")
    suspend fun getAvailableDevices(): Response<List<String>>
}